### PR TITLE
feat: add role_hierarchy check and run_in_executor deco

### PR DESCRIPTION
**Pull request by ToxicKidz**

### CODE REVIEW

File: utilities/checks.py [hunk 1]
```diff
@@ -1,22 +1,43 @@
-from discord.ext.commands import check, Context
-from json import loads
+from discord.ext.commands import check, MissingPermissions
+from functools import wraps
+from discord import Member
 
+from internal.context import Context
 
 def command_enabled(command: str):
```
Comment (anand2312) [inline utilities/checks.py:8]: I don't remember, but have we been adding return type annotations on our functions or not?

**Review by anand2312 — APPROVED** [review]
LGTM. Do we have any functions that could be using these decorators/checks? Maybe you could apply them for testing + updating codebase to use these as well on this branch before merge.

**Comment by ToxicKidz** [comment]
> LGTM. Do we have any functions that could be using these decorators/checks? Maybe you could apply them for testing + updating codebase to use these as well on this branch before merge.

Yes I've tested it on the ban command, and I think the run_in_executor one is too simple and will be tested once we start using some blocking modules.

File: utilities/checks.py (hunk 1 above)
Comment (ToxicKidz) [reply utilities/checks.py:8]: > I don't remember, but have we been adding return type annotations on our functions or not?

I don't see it too often but if you want I can add them.

Comment (anand2312) [reply utilities/checks.py:8]: I think I'd rather we maintain consistency within the project itself - if we haven't used it before, might as well not do it now.

**Pull request merged**